test/components/CollectionForm_test.js

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import CollectionForm from "../../src/components/collection/CollectionForm";
+
+const warningText =
+  "You don't have the required permission to edit this collection.";
+
+const testProps = {
+  cid: null,
+  bid: "default",
+  session: {
+    permissions: [
+      {
+        resource_name: "bucket",
+        permissions: [
+          "group:create",
+          "read",
+          "read:attributes",
+          "collection:create",
+        ],
+        bucket_id: "default",
+      },
+    ],
+    buckets: [
+      {
+        id: "default",
+      },
+    ],
+  },
+  bucket: {
+    busy: false,
+    data: {
+      id: "default",
+    },
+    permissions: {},
+    groups: [],
+  },
+  collection: {
+    busy: false,
+    data: {},
+    permissions: {},
+    records: [],
+  },
+  deleteCollection: jest.fn(),
+  onSubmit: jest.fn(),
+  formData: undefined,
+};
+
+describe("CollectionForm component", () => {
+  beforeAll(() => {
+    // preventing code mirror errors
+    Range.prototype.getBoundingClientRect = () => ({
+      bottom: 0,
+      height: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+      width: 0,
+    });
+    Range.prototype.getClientRects = () => ({
+      item: () => null,
+      length: 0,
+      [Symbol.iterator]: jest.fn(),
+    });
+  });
+
+  it("Should render an eitable form for a user with permissions creating a new collection", async () => {
```

Comment:
Nitpick: maybe set the permissions here in the test setup like in other tests with `localTestProps.session.permissions.push()` to improve readability and test atomicity (leaving global `testProps` with an empty array in permissions for example

Reply:
That's a good call.